front/app/component-library/components/CollapsibleContainer/index.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+
+import { CSSTransition } from 'react-transition-group';
+import styled from 'styled-components';
+
+import useInstanceId from '../../hooks/useInstanceId';
+import { colors, isRtl } from '../../utils/styleUtils';
+import Box, { BoxMarginProps, BoxPaddingProps, BoxWidthProps } from '../Box';
+import Icon from '../Icon';
+import Title from '../Title';
+
+type CollapsibleContainerProps = {
+  title: string;
+  children: ReactNode;
+  isOpenByDefault?: boolean;
+} & BoxMarginProps &
+  BoxWidthProps &
+  BoxPaddingProps;
+
+const ChevronIcon = styled(Icon)`
+  fill: ${colors.black};
+  margin-left: 20px;
+  transition: fill 80ms ease-out, transform 200ms ease-out;
+`;
+
+const TitleButton = styled(Box)`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  cursor: pointer;
+  width: 100%;
+  text-align: left;
+  ${isRtl`
+    text-align: right;
+    direction: rtl;
+    ${ChevronIcon} {
+        transform: rotate(180deg);
+    }
+`}
+  &.expanded {
+    ${ChevronIcon} {
+      transform: rotate(90deg);
+    }
+  }
+  &:hover {
+    ${ChevronIcon} {
+      fill: #000;
+    }
+  }
+`;
+
+const CollapseContainer = styled(Box)`
+  margin-bottom: 16px;
+  opacity: 1;
+  display: flex;
+  flex-wrap: wrap;
+  transition: 'all 1000ms cubic-bezier(0.165, 0.84, 0.44, 1)';
+  will-change: opacity, height;
+
+  ${isRtl`
+    text-align: right;
+    direction: rtl;
+`}
+
+  &.expanded-enter {
+    opacity: 0;
+    max-height: 0px;
+    overflow: hidden;
+
+    &.expanded-enter-active {
+      opacity: 1;
+      max-height: 1000px;
+      overflow: hidden;
+    }
+  }
+
+  &.expanded-enter-done {
+    opacity: 1;
+    overflow: visible;
+  }
+
+  &.expanded-exit {
+    opacity: 1;
+    max-height: 1000px;
+    overflow: hidden;
+
+    &.collapsed-exit-active {
+      opacity: 0;
+      max-height: 1000px;
+      overflow: hidden;
+    }
+  }
+`;
+
+const CollapsibleContainer = ({
+  isOpenByDefault,
+  title,
+  children,
+}: CollapsibleContainerProps) => {
+  const [isExpanded, setIsExpanded] = useState(isOpenByDefault);
+  const uuid = useInstanceId();
+
+  useEffect(() => {
+    setIsExpanded(isOpenByDefault);
+  }, [isOpenByDefault]);
```

Comment:
This should not be necessary if we use the prop already to set the default state.